### PR TITLE
Add check for spaces after commas

### PR DESCRIPTION
**Pull request by jacknoble**
Creates a new readability check SpaceAfterCommas and resolves https://github.com/rrrene/credo/issues/324 (as long as we're good with a readability check).

Let me know if I've missed something or my Elixir is rusty and I'll fix it :-)

### CODE REVIEW

File: test/credo/check/readability/space_after_commas_test.exs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+end
+""" |> to_source_file
+    |> refute_issues(@described_check)
+  end
+
+  #
+  # cases raising issues
+  #
+
+  test "it should report when commas are not followed by spaces" do
```
Comment (rrrene) [inline test/credo/check/readability/space_after_commas_test.exs]: Good test suite. :+1:

Two cases are missing:

```elixir
# should NOT trigger an issue
[?,, ?:]

# should trigger an issue
[?,,?:]
```

File: lib/credo/check/readability/space_after_commas.ex [hunk 2]
```diff
@@ -0,0 +1,56 @@
+defmodule Credo.Check.Readability.SpaceAfterCommas do
+  @moduledoc """
+  Don't use spaces after commas.
```
Comment (rrrene) [inline lib/credo/check/readability/space_after_commas.ex]: Let's write:

> You can use white-space after commas to make items of lists, tuples and other enumerations easier to separate from one another.

File: lib/credo/check/readability/space_after_commas.ex [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+      # good
+      alias Project.{Alpha, Beta}
+      arr = [1, 2, 3, 4, 5]
+      def some_func(first, second, third) do
+        ...
+      end
+
+      # hard to read
```
Comment (rrrene) [inline lib/credo/check/readability/space_after_commas.ex]: Let's write:

> NOT preferred - items are harder to separate

File: lib/credo/check/readability/space_after_commas.ex [hunk 4]
```diff
@@ -0,0 +1,56 @@
+defmodule Credo.Check.Readability.SpaceAfterCommas do
+  @moduledoc """
+  Don't use spaces after commas.
+
+      # good
```
Comment (rrrene) [inline lib/credo/check/readability/space_after_commas.ex]: Let's write:

> preferred

File: lib/credo/check/readability/space_after_commas.ex [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      end
+
+      # hard to read
+      alias Project.{Alpha,Beta}
+      arr = [1,2,3,4,5]
+      def some_func(first,second,third) do
+        ...
+      end
+
+  To make your code more readable put spaces or newlines after your commas.
```
Comment (rrrene) [inline lib/credo/check/readability/space_after_commas.ex]: There a "standard footer" for Readability issues, which we should include here:

>  Like all `Readability` issues, this one is not a technical concern.
>  But you can improve the odds of others reading and liking your code by making
>  it easier to follow.

File: lib/credo/check/readability/space_after_commas.ex [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  """
+
+  @explanation [check: @moduledoc]
+  @unspaced_commas ~r/(\,\S)/
+
+  use Credo.Check, base_priority: :low
```
Comment (rrrene) [inline lib/credo/check/readability/space_after_commas.ex]: Let's remove the `, base_priority: :low` part and make this a first-class citizen. :+1:

File: lib/credo/check/readability/space_after_commas.ex [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      arr = [1,2,3,4,5]
+      def some_func(first,second,third) do
+        ...
+      end
+
+  To make your code more readable put spaces or newlines after your commas.
+  """
+
+  @explanation [check: @moduledoc]
+  @unspaced_commas ~r/(\,\S)/
```
Comment (rrrene) [inline lib/credo/check/readability/space_after_commas.ex]: I'm pretty sure this has to be `~r/[^\?](\,\S)/`. See additional test cases below!

**Comment by jacknoble** [comment]
@rrrene Updated. Good call on `?` -- modified further because we should still warn when the comma comes after a question mark in a variable or function name and because it needs to use look behind to capture each violation.

**Comment by rrrene** [comment]
@jacknoble Great! Thx for this! :+1:

**Pull request merged**